java/arcs/core/data/Recipe.kt

```diff
@@ -40,9 +40,9 @@ data class Recipe(
         val type: Type,
         val storageKey: String? = null,
         val capabilities: Capabilities? = null,
-        val associatedHandles: List<String> = emptyList()
+        val associatedHandleNames: List<String> = emptyList(),
+        val associatedHandles: MutableList<Handle> = mutableListOf<Handle>()
```

Comment:
This should not be mutable, recipes should be immutable right now in Kotlin, I will comment below how to do this,